packages/next/server/next-server.ts

```diff
@@ -700,7 +700,9 @@ export default class Server {
         })
 
         for (let [key, value] of result.response.headers) {
-          allHeaders.append(key, value)
+          if (key !== 'x-middleware-next') {
+            allHeaders.append(key, value)
+          }
```

Comment:
Why was this change made?

It means that a middleware cannot longer set headers without breaking the middleware chain as we have below:

```js
        if (!result.response.headers.has('x-middleware-next')) {
          break
        }
```

This will break some of the examples, like [this one](https://github.com/vercel/examples/blob/30577906af4090c2031b8f18e4ecd0fcf6df1e65/edge-functions/ip-blocking-datadome/lib/datadome.ts#L133).

It also means that a middleware cannot simply add a cookie and keep the rest as is. For example, you may want to add a `csrf` cookie if missing, but not break the middleware chain or return a custom `Response`.